components/director/internal/systemfetcher/service.go

```diff
@@ -227,21 +239,75 @@ func (s *SystemFetcher) processSystemsForTenant(ctx context.Context, tenantMappi
 	return nil
 }
 
-func (s *SystemFetcher) convertSystemToAppRegisterInput(sc System) (model.ApplicationRegisterInput, string) {
-	initStatusCond := model.ApplicationStatusConditionInitial
-	baseURL := sc.BaseURL
+func (s *SystemFetcher) convertSystemToAppRegisterInput(ctx context.Context, sc System) (*model.ApplicationRegisterInputWithTemplate, error) {
+	if len(sc.TemplateID) > 0 {
+		return s.appRegisterInputFromTemplate(ctx, sc)
+	}
 
 	appRegisterInput := model.ApplicationRegisterInput{
-		Name:            sc.DisplayName,
-		Description:     &sc.ProductDescription,
-		BaseURL:         &baseURL,
-		ProviderName:    &sc.InfrastructureProvider,
-		StatusCondition: &initStatusCond,
-		SystemNumber:    &sc.SystemNumber,
-		Labels: map[string]interface{}{
-			"managed": "true",
-		},
-	}
-
-	return appRegisterInput, sc.TemplateID
+		Name: sc.DisplayName,
+	}
+
+	return enrichAppRegisterInput(appRegisterInput, sc), nil
+}
+
+func (s *SystemFetcher) appRegisterInputFromTemplate(ctx context.Context, sc System) (*model.ApplicationRegisterInputWithTemplate, error) {
+	appTemplate, err := s.appTemplateService.Get(ctx, sc.TemplateID)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while getting application template with ID %s", sc.TemplateID)
+	}
+
+	inputValues := getTemplateInputs(appTemplate, sc)
+	appRegisterInputJSON, err := s.appTemplateService.PrepareApplicationCreateInputJSON(appTemplate, inputValues)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while preparing ApplicationRegisterInput JSON from Application Template with name %s", appTemplate.Name)
+	}
+
+	appRegisterInput, err := s.appConverter.CreateInputJSONToModel(ctx, appRegisterInputJSON)
+	if err != nil {
+		return nil, errors.Wrapf(err, "while preparing ApplicationRegisterInput model from Application Template with name %s", appTemplate.Name)
+	}
+
+	return enrichAppRegisterInput(appRegisterInput, sc), nil
+}
+
+func enrichAppRegisterInput(input model.ApplicationRegisterInput, sc System) *model.ApplicationRegisterInputWithTemplate {
+	initStatusCond := model.ApplicationStatusConditionInitial
+
+	input.Name = sc.DisplayName
+	input.Description = &sc.ProductDescription
+	input.BaseURL = &sc.BaseURL
+	input.ProviderName = &sc.InfrastructureProvider
+	input.StatusCondition = &initStatusCond
+	input.SystemNumber = &sc.SystemNumber
+
+	if len(input.Labels) == 0 {
+		input.Labels = make(map[string]interface{}, 1)
+	}
+	input.Labels["managed"] = "true"
+
+	return &model.ApplicationRegisterInputWithTemplate{
+		ApplicationRegisterInput: input,
+		TemplateID:               sc.TemplateID,
+	}
+}
+
+func getTemplateInputs(t *model.ApplicationTemplate, s System) model.ApplicationFromTemplateInputValues {
+	inputValues := model.ApplicationFromTemplateInputValues{}
+	for _, p := range t.Placeholders {
+		if strings.Contains(strings.ToLower(p.Name), "url") {
```

Comment:
yep, those are the templates we have registered in our prod DBs, however, one can register whatever template with whatever placeholders, which makes the "fill-in" here a bit harder.
The dummy thing I tried to do was guess what the placeholder can actually "hold", and what we can provide, so the URL here is just a wild guess. There was also an if that checks if the placeholder contains "name", so it fills it in with the system name, but then I tought "what if its something completely different?". 
We can also do the "if contains "name"" thing, and for other cases just pass "unknown", what do you think?

Reply:
If we have any placeholder and values in the template it will be tricky/hard to map these values to the external registry values, we can think about this scenario and how to handle it. 
About the "if thing" my concern is similar to yours - I'm not sure what value we'll get from that placeholder and how will be fit in the application. If we assume the system name will be in `name` placeholder, it's okay in this case but image we have another placeholder containing `*name*` which is with completely different semantic, then we'll set  the system name on it?

P.S. These are a conceptual questions that came up to my mind when I was reviewing the PR, they may not be relevant at all, or they may not need to be handled at this point.

Reply:
You can check my answer to Dimitar below